### DISCUSSION TITLE
Use each index of an array as a useContractRead query?

### DISCUSSION THREAD

**Discussion by thekesslereffect** [wevm/wagmi#2140] [Q&A] [answered]
I am obtaining an array of token ids from a contract. I need to turn around and use that array of ids in a single argument contract read and store all the results into a new array. 
How can I do this?

**Comment by tmm** [comment, marked as answer]
Check out [`useContractReads`](https://wagmi.sh/react/hooks/useContractReads)

**Comment by thekesslereffect** [comment]
thanks @tmm with a combination of your direction and a little chatGPT magic this was the solution that worked for me:

`userGen3 = [array of tokens from previous useContractRead, blah blah blah];

const contracts = userGen3.flatMap((id) => [
    {
      address: contract_gen3token,
      abi: gen3TokenABI,
      functionName: "tokens",
      args: [id],
    }
  ])

  const { getUserGen3NFTs, data:wagmi_userGen3NFTs, isError, isLoading } = useContractReads({ contracts });
 
 useEffect(()=>{
    if(wagmi_userGen3NFTs){setUserGen3NFTs(wagmi_userGen3NFTs)}
  },[wagmi_userGen3NFTs]);`
